torchprime/torch_xla_models/train.py

```diff
@@ -175,7 +171,10 @@ def shard_output(output, mesh):
         raise ValueError(
           "Something went wrong, the output of the model shouldn't be `None`"
         )
-      xs.mark_sharding(real_output, mesh, ("fsdp", None, None))
+      # TODO: Find a better way to shard the output. It is expected that the
+      # first dimension of the output is the batch size which is usually sharded
+      # among all the devices except the tensor axis.
+      xs.mark_sharding(real_output, mesh, (("dcn", "fsdp", "expert"), None, None))
```

Comment:
File a bug plz :)

I guess I didn't know if there could even be a better way. Should this be

```
xs.mark_sharding(real_output, mesh, (("dcn", "fsdp", "expert"), None, "tensor"))
```

? Does that help?

Reply:
I meant that listing all the mesh axes to shard the output of the layer doesn't feel very intuitive. Was wondering if we can make this interface better.

Reply:
I see